### PR TITLE
Upgrade most dependencies

### PR DESCRIPTION
**Pull request by bwestlin**
### Checklist

* [x] I have read the [Contributor Guide](../../CONTRIBUTING.md)
* [x] I have read and agree to the [Code of Conduct](../../CODE_OF_CONDUCT.md)
* [x] I have added a description of my changes and why I'd like them included in the section below

### Description of Changes

Upgraded most dependencies as there are a lot of old ones, main thing I wanted to update was [octocrab](https://crates.io/crates/octocrab) as we've been seeing some recurring errors that could be helped with a newer octocrab.

I have tested running manually as well as checked that `cargo test` passes.

### CODE REVIEW

File: src/context.rs [hunk 1]
```diff
@@ -91,6 +91,7 @@ impl Client {
             .inner
             .pulls(&self.owner, repo)
             .list_reviews(pr_number)
+            .send()
```
[the remark this answers is not in the document]
Comment (bwestlin) [reply src/context.rs:94]: Yes, previously `.list_reviews()` would give an `impl Future` but it has been extended to return a Builder where e.g. paging can be configured now, so the `send` is needed to make the actual request now.

File: deny.toml [hunk 2]
```diff
@@ -1,39 +1,24 @@
+[graph]
 targets = [
     { triple = "x86_64-unknown-linux-musl" },
 ]
 
 [advisories]
-vulnerability = "deny"
-unmaintained = "deny"
+version = 2
```
Comment (bwestlin) [inline deny.toml:7]: See: https://github.com/EmbarkStudios/cargo-deny/pull/611

**Comment by bwestlin** [comment]
@vojd Updated `deny.toml` as well

**Review by davidpdrsn — APPROVED** [review]
LGTM

**Pull request merged**